### DISCUSSION TITLE
使用Github REST API,自动获取vuejs/core仓库commits

### DISCUSSION THREAD

**Discussion by FliPPeDround** [vue-mini/vue-mini#85] [Ideas]
今天在B站上看到作者的视频，需要去vusjs/core仓库里去看每一个commit，找是否有需要同步更改的文件
所以想着能不能自动查询commit，减少工作量

```js
import axios from "axios";
import fs from "fs";

const token = 'ghp_xxxxxxxxx'; // github token
const config = {
    headers: { 'Authorization': `token ${token}` }
};

axios.get('https://api.github.com/repos/vuejs/core/commits', config)
    .then(response => {
      fs.writeFileSync('commits.json', JSON.stringify(response.data, null, 2));
        console.log(response.data);
    })
    .catch(error => {
        console.error('Error:', error);
    });
```
这个脚本会获取到commit列表信息
列表信息里有url字段，访问此url可以获取该commit详情
![image](https://github.com/user-attachments/assets/4df40bf2-3337-406d-95b2-63c3dc876195)

然后检测filename是否命中所需要同步的文件，发送信息，提醒修改

可以实现一个脚本，通过git actions定时运行

**Comment by yangmingshan** [comment]
看每个 commit 并不是必要的，我这么做一是为了录视频，二是想更多的了解 Vue 的改动，比如 build 或者工作流的优化我能不能抄一抄啊之类的。仅仅是为了同步的话我去看那几个文件的提交记录就可以了，有时候提交太多不太想看时我就是这么做的 😂

**Reply by FliPPeDround** [reply]
哈哈哈，是我想多了
